src/Model.js

```diff
@@ -265,6 +266,15 @@ export class Model extends ObjectionModel {
 	}
 
 	$beforeUpdate() {
+		if(this.constructor.useIdSafeUpdates) {
+			const cols = this.constructor.getIdColumnArray()
+
+			for(let i = 0, l = cols.length; i < l; i++) {
```

Comment:
I think we should include a fast path here when there’s only a single id column, given that most cases with overwhelmingly be just `id`.  We can fall back to the for loop in cases where it’s more than one.